lib/ASTGen/Sources/ASTGen/Bridge.swift

```diff
@@ -32,12 +32,8 @@ extension BridgedSourceLoc {
     at position: AbsolutePosition,
     in buffer: UnsafeBufferPointer<UInt8>
   ) {
-    if let start = buffer.baseAddress,
-      position.utf8Offset >= 0 && position.utf8Offset < buffer.count {
-      self = SourceLoc_advanced(BridgedSourceLoc(raw: start), SwiftInt(position.utf8Offset))
-    } else {
-      self = nil
-    }
+    precondition(position.utf8Offset >= 0 && position.utf8Offset < buffer.count)
+    self = SourceLoc_advanced(BridgedSourceLoc(raw: buffer.baseAddress), SwiftInt(position.utf8Offset))
```

Comment:
I think we should force-unwrap `buffer.baseAddress`. We used to fall back to an invalid location if it was `nil` and now we're just letting it slip through.

Reply:
something like this?

```swift
extension BridgedSourceLoc {
  /// Form a source location at the given absolute position in `buffer`.
  init(
    at position: AbsolutePosition,
    in buffer: UnsafeBufferPointer<UInt8>?
  ) {
    precondition(position.utf8Offset >= 0 && position.utf8Offset < buffer!.count)
    self = SourceLoc_advanced(BridgedSourceLoc(raw: buffer!.baseAddress), SwiftInt(position.utf8Offset))
  }
}
```

Reply:
Yes

Reply:
`buffer.baseAddress` is the optional here, so @AnthonyLatsis just meant `buffer.basAddress!`